### DISCUSSION TITLE
DataHub GraphQL - Query Many to one relation

### DISCUSSION THREAD

**Discussion by nosovk** [pimcore/pimcore#10060] [Q&A / Support]
I have `object_ProductGroup` which has a one-to-many relationship to `object_ProductType`

`object_ProductType` has field slug: string.

 I want to query `object_ProductGroup` whose `object_ProductType.slug` equals to exact string.

I can create a query that takes object_ProductType.id as a filter
![image](https://user-images.githubusercontent.com/1398808/129771172-9b75b131-77bc-4dc6-b522-8721bbc0078b.png)
It works with such a filter:
```"filter": "{\"product_type__id\":{\"$like\":\"696848\"}}",```
But when I try to filter by another field everything brokes to 500
```"filter": "{\"product_type__slug\":{\"$like\":\"doors\"}}",```

'slug' field is accessible, but I can't filter over it. Are there any errors in my query? Or some additional settings to class ProductType should be done? I've added indexed and mandatory attributes to 'slug' filed, but it didn't help.

![image](https://user-images.githubusercontent.com/1398808/129772645-78fc8ddc-64a0-4d67-b897-337bcbd05150.png)

**Comment by lpheller** [comment]
Based on https://github.com/pimcore/data-hub/issues/224#issuecomment-628514018 i thought filtering based on relation attributes doesn't work at all (which would be a shame)... So I'm wondering how you have gotten `"filter": "{\"product_type__id\":{\"$like\":\"696848\"}}",` to work? I tried the same, but couldn't get it running, always getting an error like `SQLSTATE[42S22]: Column not found: 1054 Unknown column 'object_localized_11_en.product_family__id' in 'where clause'"`, though I basically used a near identical query to your screenshot.

Sorry, I currently can't help you with your original question.

**Reply by nosovk** [reply]
probably problem in object_localized_11_en - in my setup id field is a system, non localized field? And it's directly exposed in shema settings.
![image](https://user-images.githubusercontent.com/1398808/130616991-fab14247-64fc-4023-93d4-88d5c8be0d94.png)

**Reply by lpheller** [reply]
Thanks for your reply. In fact, the problem for me was that I was trying to access a many-to-many relationship without realizing, while you are querying a one-to-many relation. A one-to-many (object) relationship is stored as `attributename__id` column in the "targeted object" database table. Many-to-many relationships are stored as comma separated values in a `attributename` column. 
So for me the fix is not using `product_family__id` but `product_family`. 

But this actually might answer your question as well. You are not doing anything wrong in your query  but unfortunately the statement from https://github.com/pimcore/data-hub/issues/224#issuecomment-628514018 still is true. You can't filter or query for attributes on related objects. The pimcore graphql implementation just doesn't support it and as the "target object" table doesn't have any column named `product_type__slug` the query won't work.

**Reply by nosovk** [reply]
but if I set 'indexed' param to true for filed using in filter - shouldn't it be availible in "target object"?

**Reply by lpheller** [reply]
As far as I know, the indexing option just adds a normal database index to the object table, which can be handy and speed things up when specifically searching for the indexed value.

But this has nothing to do with the query when this object is loaded as relation. Especially since the index is placed on the "relation object" table and does not appear in the table of the "target object".

**Comment by weisswurstkanone** [comment]
Maybe have a look at
https://pimcore.com/docs/data-hub/current/GraphQL/Query/Add_Custom_Query.html
Could be a short term solution for you

**Reply by lpheller** [reply]
I find the documentation on this particular site a bit lacking. I tried my hand at it with the information this site provides, but couldn't get it running actually.

If this is a valid workaround for the otherwise missing functionality of querying relational data with GraphQL, then some more information would be super helpful.

**Reply by weisswurstkanone** [reply]
Please be a bit more specific. If you find particular points then please help us to improve the documentation by providing a PR 😄

**Comment by nosovk** [comment]
There was solution in https://github.com/pimcore/data-hub/issues/224#issuecomment-916425776
